nanoPU_sim.py

```diff
@@ -101,24 +103,14 @@ def compute_num_pkts(msg_len):
     return msg_len/Simulator.max_pkt_len if (msg_len % Simulator.max_pkt_len == 0) else msg_len/Simulator.max_pkt_len + 1
 
 def find_first_one(bitmap):
-    """Find the first set bit in the provided bitmap
+    """Find the first set bit in the provided bitmap.
+       NOTE: this may not work as expected when doing: find_first_one(~value) if value is all 1's
```

Comment:
Might want to check this to make sure it is doing what you want.

Reply:
I have done some tests on edge cases I could think of (not the all ones case) and it worked.
The reason for me to change it into this was the fact that Python stores our bitmap variables as integers, so the most significant bit is used as the sign bit which is not displayed when we use bin() function to print or operate on the value. 
Please go ahead if you thinks something else works better here. I will also search for what we can do about it.

Reply:
Let's leave it as is for now.